fairlearn/reductions/_exponentiated_gradient/exponentiated_gradient.py

```diff
@@ -73,7 +75,16 @@ def fit(self, X, y, **kwargs):
         :param y: The label vector
         :type y: numpy.ndarray, pandas.DataFrame, pandas.Series, or list
         """
-        _, y_train, sensitive_features = _validate_and_reformat_input(X, y, **kwargs)
+
+        if isinstance(self._constraints, ClassificationMoment):
+            logger.debug("Classification problem detected")
+            is_classification_reduction = True
+        else:
+            logger.debug("Regression problem detected")
```

Comment:
We don't do regression with expgrad yet, right? Won't this err out somewhere?

Reply:
No more than the code would right now surely?

Reply:
Eh. It appears that we had a bug here. If somebody tried calling ExponentiatedGradient with GroupLossMoment, there would be no complaint, but possibly a mysterious error / strange behavior down the line. I'm okay leaving this as Richard has right now. Or we could possibly raise an error right after "is_classification_reduction=False", but keep the rest of the PR including `enforce_binary_labels=is_classification_reduction`.

Reply:
I believe you're right - passing in a moment for regression would result in 'interesting' behaviour.

I can raise an exception in place of logging "Regression problem detected" but since the Stanford students should be about to start adding something to cope, unless one of you insists.....